src/utils/c_utils/src/string_c_utils.c

```diff
@@ -443,3 +443,35 @@ _Bool MallocAndSubstr(char** target, char* source, size_t len)
     *target = t;
     return true;
 }
+
+/**
+ * @brief Produces a new string where each character is replaced with output of @p mapFn call.
+ *
+ * @param src The source string (will not be mutated).
+ * @param mapFn The mapping function.
+ * @return char* The new string with characters mapped. Caller owns and must call free(). NULL is returned on failures.
+ * @details returns NULL if string is null or empty. e.g. char* mapped = ADUC_StringUtils_Map(str, tolower);
+ */
+char* ADUC_StringUtils_Map(const char* src, int (*mapFn)(int))
```

Comment:
what does mapFn take an int and not a char?

Reply:
b/c all tolower() and other such functions in ctype.h take and return int. I think it's because getc() and getchar() can return -1 (EOF) int so it interops with that type-wise.